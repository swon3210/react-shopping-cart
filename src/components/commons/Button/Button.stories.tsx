import { COLORS } from '../../../constants';
import Button, { Props } from './Button';

export default {
  title: 'Components/Commons/Button',
  component: Button,
  argTypes: {},
};

const Template = (args: Props) => <Button {...args} />;

export const Default = Template.bind({});

(Default as any).args = {
  fontColor: COLORS.WHITE,
  backgroundColor: COLORS.MINT_500,
  size: 'MD',
  children: '버튼',
};
