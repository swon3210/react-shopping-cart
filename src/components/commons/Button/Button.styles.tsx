import styled from 'styled-components';
import { COLORS } from '../../../constants';
import { Props } from './Button';

type ButtonProps = Omit<Props, 'children'>;

const BUTTON_SIZE = {
  LG: {
    fontSize: '32px',
    width: '638px',
    height: '98px',
  },
  MD: {
    fontSize: '24px',
    width: '388px',
    height: '73px',
  },
  SM: {
    fontSize: '20px',
    width: '138px',
    height: '47px',
  },
};

export const Button = styled.button<ButtonProps>(({ size, fontColor, backgroundColor, disabled }) => ({
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  color: fontColor,
  backgroundColor: disabled ? COLORS.GRAY_300 : backgroundColor,
  cursor: disabled ? 'not-allowed' : 'pointer',
  ...BUTTON_SIZE[size],
}));
