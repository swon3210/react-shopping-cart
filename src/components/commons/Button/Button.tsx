import { COLORS } from '../../../constants';
import * as Styled from './Button.styles';

export interface Props {
  children: React.ReactNode;
  /** 버튼의 크기 및 폰트 크기 유형 */
  size: 'SM' | 'MD' | 'LG';
  backgroundColor?: string;
  fontColor?: string;
  onClick?: (event: React.MouseEvent) => void;
  disabled?: boolean;
}

const Button = ({
  children,
  size = 'SM',
  onClick,
  backgroundColor = COLORS.MINT_500,
  fontColor = COLORS.WHITE,
  disabled = false,
}: Props) => {
  return (
    <Styled.Button
      disabled={disabled}
      onClick={onClick}
      size={size}
      backgroundColor={backgroundColor}
      fontColor={fontColor}
    >
      {children}
    </Styled.Button>
  );
};

export default Button;
