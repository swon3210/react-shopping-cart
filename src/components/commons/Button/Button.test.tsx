import Button from './Button';
import { render } from '@testing-library/react';

const buttonText = '버튼 텍스트';

describe('Button Component', () => {
  it('Button Snapshot', () => {
    const smallButtonUtil = render(<Button size="SM">{buttonText}</Button>);
    expect(smallButtonUtil.asFragment()).toMatchSnapshot();

    const mediumButtonUtil = render(<Button size="MD">{buttonText}</Button>);
    expect(mediumButtonUtil.asFragment()).toMatchSnapshot();

    const largeButtonUtil = render(<Button size="LG">{buttonText}</Button>);
    expect(largeButtonUtil.asFragment()).toMatchSnapshot();
  });
});
