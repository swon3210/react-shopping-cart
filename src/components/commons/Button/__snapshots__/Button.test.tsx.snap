// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button Snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdvvaa ifncFn"
    type="button"
  >
    버튼 텍스트
  </button>
</DocumentFragment>
`;

exports[`Button Component Button Snapshot 2`] = `
<DocumentFragment>
  <button
    class="sc-bdvvaa jvAQYl"
    type="button"
  >
    버튼 텍스트
  </button>
</DocumentFragment>
`;

exports[`Button Component Button Snapshot 3`] = `
<DocumentFragment>
  <button
    class="sc-bdvvaa dKrpuF"
    type="button"
  >
    버튼 텍스트
  </button>
</DocumentFragment>
`;
